lib/ansible/module_utils/network/avi/ansible_utils.py

```diff
@@ -544,13 +544,28 @@ def avi_common_argument_spec():
     Returns common arguments for all Avi modules
     :return: dict
     """
+    credentials_spec = dict(
+        controller=dict(default=os.environ.get('AVI_CONTROLLER', '')),
```

Comment:
I think you should instead use `fallback` in these cases. See https://docs.ansible.com/ansible/latest/dev_guide/developing_program_flow_modules.html#fallback

For example:
```suggestion
        controller=dict(fallback=(env_fallback, ['AVI_CONTROLLER'])),
```

(Requires `from ansible.module_utils.basic import env_fallback` above)